projects/m1/018-roulette-payouts/js/index.js

```diff
@@ -0,0 +1,86 @@
+function generateRouletteElements( specialValues, lowestValue, highestValue ){
+    let elementsList = [];
+
+    elementsList = elementsList.concat( specialValues );
+
+    for( i = lowestValue; i < highestValue; i++ ){
+            elementsList.push( i );
+    }
+
+    return elementsList;
+}
+
+function spinRoulette( rouletteElements ){
+    const extractedValue = Math.floor(Math.random() * rouletteElements.length);
+    return rouletteElements[extractedValue];
+}
+// I prefer to create this function instead using "includes" built-in function (like in previous exercies) to have better compatibility.
+function isInArray( element, array ){
+    if( array.indexOf(element) !== -1 ){
+        return true;
+    }
+     
+    return false;
+}
+
+function checkBet( extractedValueToCheck, reds, greens ){
+    let parity = '';
+    let range = '';
+    let color = 'Black';
+    
+    if( isInArray( extractedValueToCheck, greens ) ){
+        color = 'Green';
+    } else {
+        parity = 'Even';
+        range = '1 to 18';
+
+        if( isInArray( extractedValueToCheck, reds ) ){
+            color = 'Red';
+        }
+
+        if ( extractedValueToCheck % 2 != 0 ){
+            parity = 'Odd';
+        }
+
+        if ( extractedValueToCheck > 18 ){
+            range = '19 to 36';
+        }
+    }
+
+    return [extractedValueToCheck, color, parity, range];
+}
+
+function analyzeBet( bet ){
+
+    const [ extractedValueNumber, extractedValueColor, extractedValueParity, extractedValueRange ] = bet;
+
+    let message = 'The spin resulted in ' + extractedValueNumber + '\n';
+    message += 'Pay ' + extractedValueNumber;
+
+    if( extractedValueColor !== 'Green' ){
+        message += '\n';
+        message += 'Pay ' + extractedValueColor + '\n';
+        message += 'Pay ' + extractedValueParity + '\n';
+        message += 'Pay ' + extractedValueRange;
+    }

```

Comment:
Front End Style :-)

Reply:
It might be useful for multiline strings:
https://www.freecodecamp.org/news/javascript-multiline-string-how-to-create-multi-line-strings-in-js/